include/forcing/ForcingsEngineDataProvider.hpp

```diff
@@ -0,0 +1,65 @@
+#ifndef NGEN_FORCING_FE_PER_FEATURE_DATA_PROVDER_HPP
+#define NGEN_FORCING_FE_PER_FEATURE_DATA_PROVDER_HPP
+
+#include "GenericDataProvider.hpp"
+
+namespace data_access {
+
+struct ForcingsEngine;
+
+struct ForcingsEngineDataProvider
+  : public GenericDataProvider
+{
+
+    ForcingsEngineDataProvider() = default;
+
+    ForcingsEngineDataProvider(
+        const std::string& init,
+        std::size_t time_start,
+        std::size_t time_end
+    );
+
+    ForcingsEngineDataProvider(
+        const std::string& init,
+        const std::string& time_start,
+        const std::string& time_end,
+        const std::string& time_fmt = "%Y-%m-%d %H:%M:%S"
+    );
+
+    ~ForcingsEngineDataProvider() override;
+
+    /**
+     * @brief Set the MPI communicator that the Forcings Engine should use.
+     *
+     * @param handle int representing a communicator handle.
+     */
+    void set_communicator(int handle);
```

Comment:
Before I really dig into `ForcingsEngine`, I think it's probably preferable for that to live in its own header, and for this to be a method only exposed on that and not the `Provider`. There's a weird side-effect of calling this on one `Provider` instance affecting all of them.